c/src/ml-api-inference-single.c

```diff
@@ -1111,6 +1115,55 @@ ml_single_open_full (ml_single_h * single, const char *model,
   return ml_single_open_custom (single, &info);
 }
 
+/**
+ * @brief Open new single handle with given option.
+ */
+int
+ml_single_open_with_option (ml_single_h * single, ml_option_h option)
+{
+  ml_option_s *_option = (ml_option_s *) option;
```

Comment:
let's move these under line 1141, after checking the null ptr.
```
ml_option_s *_option;
GHashTable *table;
...
_option = (ml_option_s *) option;
table = _option->option_table;
```
(If option is null, below 1125 will crash the app.)

Reply:
Also you can add negative testcase w/ ml_single_open_with_option (single, null); :)

Reply:
Thanks! I fixed the possible segfault and added a new simple negative tc. PTAL